v0/destinations/personalize/transform.js

```diff
@@ -29,62 +34,57 @@ async function process(event) {
         !(
           keyField[k].toUpperCase() === "USER_ID" ||
           keyField[k].toUpperCase() === "EVENT_TYPE" ||
-          keyField[k].toUpperCase() === "TIMESTAMP"
+          keyField[k].toUpperCase() === "TIMESTAMP" ||
+          keyField[k].toUpperCase() === "ITEM_ID" ||
+          keyField[k].toUpperCase() === "EVENT_VALUE" ||
+          keyField[k].toUpperCase() === "IMPRESSION" ||
+          keyField[k].toUpperCase() === "RECOMMENDATION_ID" ||
+          keyField[k].toUpperCase() === "EVENT_ID"
         )
       ) {
         const mappedFields = mappedField[k];
-
         keyField[k] = keyField[k]
           .toLowerCase()
-          .replace(/_([a-z])/g, function (g) {
+          .replace(/_([a-z])/g, function camelCase(g) {
             return g[1].toUpperCase();
           });
-        if (message.properties && message.properties[mappedFields]) {
-          property[keyField[k]] = message.properties[mappedFields];
+        if (properties && properties[mappedFields]) {
+          property[keyField[k]] = properties[mappedFields];
         } else if (typeof message[mappedFields] !== "undefined") {
```

Comment:
no need of this

Reply:
are you talking about line no 46 & 47? without that the field names won't be converted to camelCase.

Reply:
or you are talking about line no 53

Reply:
This comment is on line 53.we only check under properties.